quickjs/src/main/jni/quickjs-jni.c

```diff
@@ -0,0 +1,67 @@
+#include <string.h>
+#include <jni.h>
+#include <malloc.h>
+#include "quickjs/quickjs.h"
+
+JNIEXPORT jlong JNICALL
+Java_com_squareup_quickjs_QuickJs_createContext(JNIEnv *env, jclass type) {
+  JSRuntime* runtime = JS_NewRuntime();
+  JSContext* context = JS_NewContext(runtime);
+  return (jlong) context;
+}
+
+JNIEXPORT void JNICALL
+Java_com_squareup_quickjs_QuickJs_destroyContext(JNIEnv *env, jclass type, jlong context_) {
+  JSContext* context = (JSContext*) context_;
+  JSRuntime* runtime = JS_GetRuntime(context);
+  JS_FreeContext(context);
+  JS_FreeRuntime(runtime);
+}
+
+JNIEXPORT jstring JNICALL
+Java_com_squareup_quickjs_QuickJs_evaluate__JLjava_lang_String_2Ljava_lang_String_2(JNIEnv *env,
+                                                                                    jclass type,
+                                                                                    jlong context_,
+                                                                                    jstring sourceCode_,
+                                                                                    jstring fileName_) {
+  JSContext* context = (JSContext*) context_;
+
+  const char *sourceCode = (*env)->GetStringUTFChars(env, sourceCode_, 0);
+  const char *fileName = (*env)->GetStringUTFChars(env, fileName_, 0);
+
+  JSValue evalValue = JS_Eval(context, sourceCode, strlen(sourceCode), fileName, 0);
```

Comment:
no strlen of fileName?

Reply:
Maybe because you want to be able to evaluate a subset? But no, it's not.